dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/JNDITest.java

```diff
@@ -0,0 +1,106 @@
+package io.openliberty.checkpoint.fat;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.websphere.simplicity.ShrinkHelper.DeployOptions;
+import com.ibm.websphere.simplicity.config.JNDIEntry;
+import com.ibm.websphere.simplicity.config.ServerConfiguration;
+
+import componenttest.annotation.Server;
+import componenttest.annotation.SkipIfCheckpointNotSupported;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.HttpUtils;
+import io.openliberty.checkpoint.spi.CheckpointPhase;
+import jndiApp.JNDIservlet;
+
+@RunWith(FATRunner.class)
+@SkipIfCheckpointNotSupported
+public class JNDITest {
+
+    public static final String APP_NAME = "jndiApp";
+
+    @Server("jndiServer")
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void copyAppToDropins() throws Exception {
+        server.deleteAllDropinApplications();
+        WebArchive webappWar = ShrinkWrap.create(WebArchive.class, "JNDIApplication.war").addClass(JNDIservlet.class);
+        ShrinkHelper.exportAppToServer(server, webappWar, DeployOptions.OVERWRITE);
+    }
+
+    @Test
+    public void testJNDIlookupNoEntry() throws Exception {
+
+        server.setCheckpoint(CheckpointPhase.APPLICATIONS);
+        server.startServer();
+
+        ServerConfiguration config = server.getServerConfiguration();
+        JNDIEntry defaultEntry = new JNDIEntry();
+        defaultEntry.setJndiName("jndi/value");
+        defaultEntry.setValue("default value");
+        config.getJndiEntryElements().add(defaultEntry);
+        server.updateServerConfiguration(config);
+
+        server.stopServer();
+
+        server.checkpointRestore();
+
+        HttpUtils.findStringInUrl(server, "jndiApp/servlet", "default value");
+    }
+
+    @Test
+    public void testJNDIlookup() throws Exception {
+
+        ServerConfiguration preConfig = server.getServerConfiguration();
+        JNDIEntry defaultEntry = new JNDIEntry();
+        defaultEntry.setJndiName("jndi/value");
+        defaultEntry.setValue("default value");
+        defaultEntry.setId("jndiEntry");
+        preConfig.getJndiEntryElements().add(defaultEntry);
+        server.updateServerConfiguration(preConfig);
+
+        server.startServer();
+        HttpUtils.findStringInUrl(server, "jndiApp/servlet", "default value");
+        server.stopServer();
+
+        ServerConfiguration config = server.getServerConfiguration();
+        config.getJndiEntryElements().getById("jndiEntry").setValue("alternate value");
+        server.updateServerConfiguration(config);
+
+        server.setCheckpoint(CheckpointPhase.APPLICATIONS, false, null);
```

Comment:
You should set the checkpoint before starting the server the first time where we validate that the "default value" is being used.  That way wen you start the server again after the config change it will test that it detects the updated "alternate value" on restore.